src/proteus/dswx_hls.py

```diff
@@ -4368,7 +4375,15 @@ def _check_ancillary_inputs(check_ancillary_inputs_coverage,
 
         flag_error = False
         # Handle antimeridian ("dateline") crossing
-        if file_srs.IsGeographic() and tile_min_x < 180 and tile_max_x >= 180:
+        #
+        # The dateline crossing by the tile is tested by the condition:
+        #     `tile_min_x < 180 and tile_max_x >= 180`
+        # The ancillary input only requires special handling if it is in
+        # geographic coordinates and if its longitude domain is represented
+        # within the [-180, +180] range, verified by the test `min_x < -170`,
+        # rather than the [0, +360] interval.
+        if (file_srs.IsGeographic() and min_x < -170 and
+                tile_min_x < 180 and tile_max_x >= 180):
```

Comment:
This looks like code duplication. Please pull this logic out into a standalone function. Please have that function use simplified logic akin to the suggestion in my previous comment, and then call that function in both places.

The dateline crossing issue has been a meaningful concern for SDS et al, and thus might have further tweaks in the future. If so, then it would be extremely easy for a maintainer coming in 5 months from now to update the "check" conditions in one location of the code but not the other, but think they'd completed the bug fix.

Thanks!

Reply:
Thank you, @nemo794 . Please keep in mind that a new function requires docstrings and retesting. We need a quick fix and make sure everything works.

Reply:
Done